Docker restructuring

Nazgolze opened the pull request:
This is a proposed change - looks bigger than it is :sweat_smile:.

- Moves all of the docker stuff to the root level of the folder
- Modifies Dockerfile to hardcode liskjs username, leaves $UID stuff intact 
- Simplifies docker-compose.yml, mounts entire lisk-sdk-examples directory to /home/liskjs in container
- moved create_dev_env to workshop-example
- copied node/package* to workshop-example

The README.md that used to be in create_dev_env is now DOCKER.md in the root folder.
The instructions would have to be modified slightly (already edited the `docker build` line)

This should be idempotent

Nazgolze committed:
Docker restructuring

Nazgolze committed:
Merge branch 'lisk-transport' into lisk-transport-docker-change

Nazgolze commented:
Addresses #43

fchavant commented on Dockerfile:
```diff
@@ -2,18 +2,15 @@ FROM ubuntu:18.04
 RUN apt-get update && apt-get install --assume-yes libtool automake autoconf curl python-minimal build-essential git vim || true
 RUN apt-get install --assume-yes sudo postgresql-client-10
-RUN echo "$user_name ALL=(ALL) NOPASSWD: ALL" > /etc/sudoers.d/$user_name
+RUN echo "liskjs ALL=(ALL) NOPASSWD: ALL" > /etc/sudoers.d/liskjs
```
What is this used for BTW?

Nazgolze replied:
in case the user is inside of the container and wants to sudo and install anything else inside of the container

Nazgolze replied:
will remove this, it's not necessary

fchavant commented on DOCKER.md:
```diff
@@ -20,7 +20,7 @@ cd create-dev-env`
 
```
The previous block needs to be removed

fchavant commented on DOCKER.md:
````diff
@@ -20,7 +20,7 @@ cd create-dev-env`
 To get started, create your docker image:
 ```bash
-docker build -t lisk-sdk --build-arg user_name=liskjs --build-arg user_id=$UID .
+docker build -t lisk-sdk --build-arg user_id=$UID .
````
What's the reason the `Dockerfile` does not derive `FROM nodejs`? (it would build in a couple seconds instead of minutes)

Nazgolze replied:
Yeah, I think it should.

The original incarnation of this was more of an explicit "here's everything that it's doing" thing and trying to be very minimal.

fchavant commented on DOCKER.md:
````diff
@@ -20,7 +20,7 @@ cd create-dev-env`
+docker build -t lisk-sdk --build-arg user_id=$UID .
 ```
 
 Now bring everything up using the Docker Compose file:
````
The `docker exec [...]` below should be `docker-compose exec sdk [...]`

fchavant commented on DOCKER.md:
````diff
@@ -76,4 +76,4 @@ You can simply take down the containers by executing the following command in th
 
 ```sh
 docker-compose down
````
This will leave a `lisk-sdk-examples_db-data` docker volume behind.

fchavant commented on Dockerfile:
```diff
@@ -2,18 +2,15 @@ FROM ubuntu:18.04
 
-RUN useradd --create-home --uid $user_id $user_name
+RUN useradd --create-home --uid $user_id liskjs
 RUN apt-get update && apt-get install --assume-yes libtool automake autoconf curl python-minimal build-essential git vim || true
```
@michielmulders  this `|| true` might cause problems; why no fail fast?

Nazgolze committed:
Updated documentation related to getting into the container
Dockerfile updated to use node:10-stretch
Changed to using node user, whose uid gets updated as needed

Nazgolze commented:
My latest commit should address all of the above points, except for https://github.com/LiskHQ/lisk-sdk-examples/pull/44#discussion_r343531659

(edit: and except for this too: https://github.com/LiskHQ/lisk-sdk-examples/pull/44#discussion_r343529365)

Nazgolze committed:
Some documentation corrections in DOCKER.md

fchavant commented on DOCKER.md:
````diff
@@ -76,4 +70,4 @@ You can simply take down the containers by executing the following command in th
 
 ```sh
 docker-compose down
````
```suggestion
docker-compose down --volumes
```

Nazgolze committed:
Update `docker-compose down` in DOCKER.md to remove db volume
